provider/marathon/marathon.go

```diff
@@ -284,7 +312,13 @@ func (p *Provider) getPort(task marathon.Task, application marathon.Application)
 	return strconv.Itoa(port)
 }
 
-func (p *Provider) getWeight(application marathon.Application) string {
+// Extract weight from labels for a given service and a given marathon application
+func (p *Provider) getWeight(application marathon.Application, varg ...string) string {
+	serviceName := extractServiceName(varg...)
+	if value, ok := getApplicationServiceLabel(application, serviceName, "weight"); ok {
+		return value
+	}
+
 	if label, ok := p.getLabel(application, types.LabelWeight); ok {
```

Comment:
Ah true, we already have a custom implementation. Sure, please go ahead and extend that one.